patterns/callback/README.md

```diff
@@ -0,0 +1,24 @@
+# Callback Pattern
+
+This is an example of callback patterns in Aztec.
+
+In this example, the callback effect is to set a value note for the original sender, Alice.
+
+Two types of callback are demonstrated: atomic and asynchronous.
+
+There are 2 contracts and 2 parties involved:
+- Alice: initiate the flow and receive the effect of the callback
+- Bob: trigger the logic on the callbacker contract, if the callback is async (see below)
+- call_me_back contract: initiate the flow, calls callbacker and receive the callback
+- callbacker contract: either store a pending callback (later resolved by bob) or trigger the callback immediately (if the callback is atomic)
+
+- The atomic callback flow is as follow:
+Alice calls new_atomic_call on call_me_back contract, which calls call_me_now on the resolver contract, which calls the callback function on call_me_back contract.
+
+- The async callback flow is as follow:
+Alice calls new_async_call on call_me_back contract, which calls queue_new_call on the resolver contract and stores the pending callback (in a note for Bob).
+Later, Bob calls resolve on the resolver contract, which calls the callback function on call_me_back contract.
+
+
+### Real world example
+- [Private oracle](https://github.com/defi-wonderland/aztec-private-oracle): this oracle implementation provides an extra field for an optional callback which is triggered when the oracle is updated (async callback).
```

Comment:
I would make this more explicit because I think people will look at this. Something like this:
```suggestion
- [Private oracle](https://github.com/defi-wonderland/aztec-private-oracle): This oracle implementation uses the async callback pattern. A user can ask a question to the oracle and that question is shared with an address called `divinity`. That `divinity` will receive a note with the question, and when it has the answer it will call `submit_answer` to the oracle. 
This function, in turn, will trigger a callback to the contract where the question came from. By doing this, it allows the contract to make use of that answer.```